### DISCUSSION TITLE
[gatsby-plugin-styled-components] Disable displayName option only in production build

### DISCUSSION THREAD

**Discussion by AnsonH** [gatsbyjs/gatsby#32867] [Help] [answered]
In the [gatsby-plugin-styled-components][1] plug-in options, there's a field called [`displayName`][2]. When it's set to true, it enhances the attached CSS class name on each component such that you can see something like `<button class="Button-asdf123 asdf123" />` instead of `<button class="asdf123" />` in DevTools.

However, how can I set up this option such that `displayName` is true ONLY in development server (`gatsby develop`), but not in production build (`gatsby build`)? In other words, I want to set `displayName` to true during development and false during building for production.

Someone asked [similar question on Stack Overflow][3]. However, I'm not sure how I can do it in Gatsby. Thanks!

[1]: https://www.gatsbyjs.com/plugins/gatsby-plugin-styled-components/#options
[2]: https://styled-components.com/docs/tooling#better-debugging
[3]: https://stackoverflow.com/q/55546915/11067496

**Comment by herecydev** [comment, marked as answer]
Can you not leverage `process.env.NODE_ENV`, i.e.

```
{
displayName: process.env.NODE_ENV === "development" ? true : false
}
```
